src/Led.cpp

```diff
@@ -224,500 +255,566 @@ static void Led_Task(void *parameter) {
+											} else {
+												animationIndex = 42;
+												subAnimationIndex = staticLastBarLenghtPlaylist;
+											}
+										}
+									}
+									staticLastTrack = gPlayProperties.currentTrackNumber;
 								}
-						} else {
-							if (Wlan_ConnectionTryInProgress()) {
-								idleColor = CRGB::Orange;
+
+								if (animationIndex == 0 ) {
+									animationActive = true;
+									subAnimationIndex = 0;
+									animationIndex++;
+								}
+
+								animaitonTimer = 30;
+								uint8_t barLength = 0;
+								switch (animationIndex) {
+									case 1:
+										barLength = subAnimationIndex;
+										subAnimationIndex++;
+										if (subAnimationIndex >= fullLeds){
+											animationIndex++;
+											subAnimationIndex = 0;
+										} 
+									break;
+									case 2:
+										// wait
+										subAnimationIndex ++;
+										if (subAnimationIndex >= 50){
+											animationIndex++;
+											subAnimationIndex = fullLeds;
+										}
+										barLength = fullLeds;
+									break;
+									case 3:
+										// negative
+										subAnimationIndex--;
+										barLength = subAnimationIndex;
+										if (subAnimationIndex == 0){
+											animationIndex++;
+										}
+									break;
+									case 42:
+										// move back to target and wait there
+										subAnimationIndex--;
+										barLength = subAnimationIndex;
+										if (subAnimationIndex <= fullLeds){
+											animationIndex = 2;
+										}
+									break;
+									default:
+										// done
+										animationActive = false;
+									break;
+								}
+					
+								// draw bar
+								FastLED.clear();
+								for (uint8_t i = 0; i < barLength; i++){
+									leds[Led_Address(i)] = CRGB::Blue;
+								}
+								if (barLength == fullLeds && lastLed > 0){
+									leds[Led_Address(barLength)] = Led_DimColor(CRGB::Blue, lastLed);
+								}
+								staticLastBarLenghtPlaylist = barLength;
+								FastLED.show();
+							}
+						}
+					} break;
+
+					// --------------------------------------------------
+					// Animation of Idle-State
+					// --------------------------------------------------
+					case LedAnimationType::Idle: {
+						animationActive = true;
+						if (animationIndex < NUM_LEDS) {
+							if (OPMODE_BLUETOOTH_SINK == System_GetOperationMode()) {
+								idleColor = CRGB::Blue;
+							} else if (OPMODE_BLUETOOTH_SOURCE == System_GetOperationMode()) {
+								if (Bluetooth_Source_Connected()) {
+										idleColor = CRGB::Blue;
+									} else {
+										idleColor = CRGB::BlueViolet;
+									}
 							} else {
-								if (Wlan_IsConnected() && gPlayProperties.currentSpeechActive) {
-									idleColor = speechColor;
+								if (Wlan_ConnectionTryInProgress()) {
+									idleColor = CRGB::Orange;
 								} else {
-									if (Wlan_IsConnected()) {
-										idleColor = CRGB::White;
+									if (Wlan_IsConnected() && gPlayProperties.currentSpeechActive) {
```

Comment:
This if still feels a bit convoluted. Something like this could be better readale

```cpp
// Bluetooth modes and so on [....]
// row 697
} else {
	idleColor = CRGB::Green;
	if (Wlan_IsConnected()) {
		idleColor = CRGB::White;
		if(gPlayProperties.currentSpeechActive) {
			idleColor = speechColor;
		}
	}
}
```